synapseclient/credentials/cred_data.py

```diff
@@ -0,0 +1,55 @@
+from collections import namedtuple
+
+import synapseclient.utils
+import time
+import base64, hmac, hashlib
+
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
+
+
+class SynapseCredentials(object): #TODO: inherit requests.AuthBase so that this object can be simply passed to requests library
+    """
+    Credentials used to make requests to Synapse.
+    """
+
+    #setting and getting api_key it will use the base64 encoded string representation
+    @property
+    def api_key(self):
+        return base64.b64encode(self._api_key).decode()
+
+    @api_key.setter
+    def api_key(self, value):
+        self._api_key = base64.b64decode(value)
+
+    def __init__(self, username, api_key_string):
+        self.username = username
+        self.api_key = api_key_string
+
+    def get_signed_headers(self, url):
+        """
+        Generates signed HTTP headers for accessing Synapse urls
+        :param url:
+        :return:
+        """
+        sig_timestamp = time.strftime(synapseclient.utils.ISO_FORMAT, time.gmtime())
```

Comment:
a note to consider: can one uses python client to make anonymous calls? for example, getting synapse version, getting a list of synapse users, or viewing public entity, ...